asmcomp/arm/emit.mlp

```diff
@@ -74,13 +74,18 @@ let slot_offset env loc cl =
   | Outgoing n ->
       assert (n >= 0);
       n
+  | Domainstate _ -> assert false  (* not a stack slot *)
 
 (* Output a stack reference *)
 
 let emit_stack env r =
   match r.loc with
+  | Stack (Domainstate n) ->
+      let ofs = n + Domainstate.(idx_of_field Domain_extra_params) * 8 in
```

Comment:
```suggestion
      let ofs = n + Domainstate.(idx_of_field Domain_extra_params) * 4 in
```

Reply:
No. Even on 32-bit platforms, entries in the domain state are 8 bytes wide.  

Note for later: introduce a `Domainstate.offset_of_field` function that does the `* 8`.

Reply:
Ah, of course. Somewhere in the middle of reviewing I thought this was actually computing an offset in the `extra_params` array.